### DISCUSSION TITLE
Fastlane recompiles SPM dependencies every time

### DISCUSSION THREAD

**Discussion by mmdumi** [fastlane/fastlane#20375] [Q&A]
Hi,
My Fastfile build lane looks like. The issue I'm seeing is that even on the CI platform or locally fastlane always recompiles the SPM dependencies. This increases my build time drastically, so I would like to ask how should I configure fastlane to preserve previous build caches for spm dependencies.

```
lane :build_beta do |options|
    build_app(
      project: "TestCircleci/TestCircleci.xcodeproj",
      clean: false,
      cloned_source_packages_path: "SourcePackages"
    )
  end
```

I've seen this issue from trying to compile my project in a CI system (circleci) where my builds are getting larger and larger if I'm adding project dependencies with SPM (swift package manager) instead of regular prebuild frameworks: .xcframework.

For example this really basic project with a single dependency:
- compiles in 16m with a SPM dependency because it compiles it every time no matter the caching system
- compiles in 7m with a .xcframework dependency

**Comment by dokimyj** [comment]
https://forums.swift.org/t/swiftpm-with-git-lfs/42396

I asked the professionals in Swift Official Forum for this question.

In short, there is no complete solution to cache the Swift Package Manager dependencies.

(They said we gotta make 'Dynamic Library' for the chunk of SwiftPM dependencies in other package management tools like Carthage.)
